File: scala-finch/src/main/scala/org/openapitools/models/BadEventQueueIdError.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import org.openapitools.models.AnyType
import org.openapitools.models.BadEventQueueIdErrorAllOf
import org.openapitools.models.CodedErrorBase

/**
 * 
 * @param result 
 * @param msg 
 * @param code 
 * @param queueUnderscoreid The string that identifies the invalid event queue. 
 */
case class BadEventQueueIdError(result: AnyType,
                msg: AnyType,
                code: Option[AnyType],
                queueUnderscoreid: Option[String]
                )

object BadEventQueueIdError {
    /**
     * Creates the codec for converting BadEventQueueIdError from and to JSON.
     */
    implicit val decoder: Decoder[BadEventQueueIdError] = deriveDecoder
    implicit val encoder: ObjectEncoder[BadEventQueueIdError] = deriveEncoder
}

File: fsharp-functions/OpenAPI/src/model/BadEventQueueIdError.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open Newtonsoft.Json
open OpenAPI.Model.AnyType
open OpenAPI.Model.BadEventQueueIdErrorAllOf
open OpenAPI.Model.CodedErrorBase

module BadEventQueueIdError = 

  //#region BadEventQueueIdError

  [<CLIMutable>]
  type BadEventQueueIdError = {
    [<JsonProperty(PropertyName = "result")>]
    Result : AnyType;
    [<JsonProperty(PropertyName = "msg")>]
    Msg : AnyType;
    [<JsonProperty(PropertyName = "code")>]
    Code : AnyType;
    [<JsonProperty(PropertyName = "queue_id")>]
    QueueId : string;
  }
  
  //#endregion
  